utilities/write_batch_with_index/write_batch_with_index_internal.cc

```diff
@@ -307,7 +312,8 @@ void BaseDeltaIterator::UpdateCurrent() {
         if (comparator_->CompareWithoutTimestamp(
                 delta_entry.key, /*a_has_ts=*/false, *iterate_upper_bound_,
                 /*b_has_ts=*/false) >= 0) {
-          // out of upper bound -> finished.
+          // out of upper bound -> delta finished.
+          delta_iterator_out_of_bound_ = true;
```

Comment:
Should we reset `delta_iterator_out_of_bound_` to false when a user calls Seek* again?

Reply:
Good point... we should probably pass the upper_bound down to the WBWIIteratorImpl and have it do the upper-bound check normally, i.e., by returning `Valid() == false` upon reaching the upper-bound.